src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorObject.cs

```diff
@@ -0,0 +1,311 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkWatcherConnectionMonitorObject", SupportsShouldProcess = true), OutputType(typeof(PSNetworkWatcherConnectionMonitorObject))]
+    public class AzureNetworkWatcherConnectionMonitorObjectCommand : ConnectionMonitorBaseCmdlet
+    {
+        [Parameter(
+             Mandatory = true,
+             HelpMessage = "The network watcher resource.",
+             ParameterSetName = "SetByResource")]
+        [ValidateNotNull]
+        public PSNetworkWatcher NetworkWatcher { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of network watcher.",
+            ParameterSetName = "SetByName")]
+        [ResourceNameCompleter("Microsoft.Network/networkWatchers", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string NetworkWatcherName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the network watcher resource group.",
+            ParameterSetName = "SetByName")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Location of the network watcher.",
+            ParameterSetName = "SetByLocation")]
+        [LocationCompleter("Microsoft.Network/networkWatchers/connectionMonitors")]
+        [ValidateNotNull]
+        public string Location { get; set; }
+  
+        [Alias("ConnectionMonitorName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The list of test group.")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNetworkWatcherConnectionMonitorTestGroupObject> TestGroup { get; set; }
```

Comment:
For collection types, you should use arrays as the parameter type, i.e. ```PSNetworkWatcherConnectionMonitorTestGroupObject[]```

Reply:
using List mnake it eays to use *.NET and LINQ functions.